src/vs/workbench/contrib/terminalContrib/chat/browser/terminal.chatHint.contribution.ts

```diff
@@ -0,0 +1,263 @@
+import { TerminalChatCommandId } from 'vs/workbench/contrib/terminalContrib/chat/browser/terminalChat';
+import { TerminalInstance } from 'vs/workbench/contrib/terminal/browser/terminalInstance';
+const $ = dom.$;
+
+class TerminalChatHintContribution extends Disposable implements ITerminalContribution {
+	static readonly ID = 'terminal.chatHint';
+
+	private _hintWidget: HTMLElement | undefined;
+
+	static get(instance: ITerminalInstance | IDetachedTerminalInstance): TerminalChatHintContribution | null {
+		return instance.getContribution<TerminalChatHintContribution>(TerminalChatHintContribution.ID);
+	}
+
+	private _xterm: IXtermTerminal & { raw: RawXtermTerminal } | undefined;
+	private _chatHint: IDecoration | undefined;
+
+	constructor(
+		private readonly _instance: ITerminalInstance | IDetachedTerminalInstance,
+		processManager: ITerminalProcessManager | ITerminalProcessInfo,
+		widgetManager: TerminalWidgetManager,
+		@IInlineChatService private readonly _inlineChatService: IInlineChatService,
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@ITerminalService private readonly _terminalService: ITerminalService
+	) {
+		super();
+	}
+
+	xtermOpen(xterm: IXtermTerminal & { raw: RawXtermTerminal }): void {
+		if (this._terminalService.instances.length !== 1) {
+			// only show for the first terminal
+			return;
+		}
+		this._xterm = xterm;
+		const capability = this._instance.capabilities.get(TerminalCapability.CommandDetection);
+		if (capability) {
+			this._register(Event.once(capability.promptInputModel.onDidStartInput)(() => this._addChatHint()));
+		} else {
+			this._register(this._instance.capabilities.onDidAddCapability(e => {
+				if (e.capability.type === TerminalCapability.CommandDetection) {
+					const capability = this._instance.capabilities.get(TerminalCapability.CommandDetection);
+					this._register(Event.once(capability!.promptInputModel.onDidStartInput)(() => this._addChatHint()));
+					if (!capability!.promptInputModel.value) {
+						this._addChatHint();
+					}
+				}
+			}));
+		}
+
+		this._register(Event.once(this._inlineChatService.onDidChangeProviders)(() => this._addChatHint()));
+	}
+
+	private _addChatHint(): void {
+		const instance = this._instance instanceof TerminalInstance ? this._instance : undefined;
+		if (!instance || !this._xterm || this._hintWidget || instance?.capabilities.get(TerminalCapability.CommandDetection)?.hasInput) {
+			return;
+		}
+
+		if (!this._chatHint) {
+			const marker = this._xterm.raw.registerMarker();
+			if (!marker) {
+				return;
+			}
+
+			if (this._xterm.raw.buffer.active.cursorX === 0) {
+				return;
+			}
+			this._register(marker);
+			this._chatHint = this._xterm.raw.registerDecoration({
+				marker,
+				x: this._xterm.raw.buffer.active.cursorX + 1
+			});
+		}
+
+		this._register(this._xterm.raw.onKey(() => this._chatHint?.dispose()));
```

Comment:
Include `onCursorMove` here would cover our bases (eg. sending text from some other component). Something to be careful of is focusing the terminal map send data through the pty so `onData` won't work.

Reply:
`onCursorMove` happens when the prompt comes in, so that doesn't work 🤔